src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -386,6 +376,26 @@ private function addServiceToAmbiguousType($id, $type)
         $this->ambiguousServiceTypes[$type][] = $id;
     }
 
+    /**
+     * @param \ReflectionClass $reflectionClass
+     *
+     * @return \ReflectionMethod[]
+     */
+    private static function getSetters(\ReflectionClass $reflectionClass)
+    {
+        $setters = array();
+        foreach ($reflectionClass->getMethods(\ReflectionMethod::IS_PUBLIC) as $reflectionMethod) {
+            $name = $reflectionMethod->getName();
+            if ($reflectionMethod->isStatic() || 1 !== $reflectionMethod->getNumberOfParameters() || 0 !== strpos($name, 'set')) {
+                continue;
+            }
+
+            $setters[] = $reflectionMethod;
```

Comment:
instead of building an array, you could yield the ReflectionMethod, which would keep the loop lazy...